shard-operator-parent/shard-operator-core/src/main/java/com/redhat/service/smartevents/shard/operator/core/resources/CustomResourceStatus.java

```diff
@@ -45,6 +45,11 @@ public final boolean isReady() {
         return conditions.stream().anyMatch(c -> ConditionTypeConstants.READY.equals(c.getType()) && ConditionStatus.True.equals(c.getStatus()));
     }
 
+    @JsonIgnore
+    public final boolean allConditionsAreReady() {
```

Comment:
This name may be misleading.. 

For "deletion" of resources we have a Condition named `BridgeDeleted` that can be `True` but it is not "ready".